plugins/modules/cloudwatch_metric_alarm_info.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/python
+# This file is part of Ansible
+# GNU General Public License v3.0+ (see COPYING or https://wwww.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: cloudwatch_metric_alarm_info
+version_added: 5.0.0
+short_description: Gather information about the alarms for the specified metric
+description:
+  - Retrieves the alarms for the specified metric.
+author:
+  - Mandar Vijay Kulkarni (@mandar242)
+options:
+  alarm_names:
+    description:
+      - The name of the metric.
+    required: false
+    type: list
+    elements: str
+
+extends_documentation_fragment:
+- amazon.aws.aws
+- amazon.aws.ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: describe the metric alarm based on metric name and namespace
+  amazon.aws.cloudwatch_metric_alarm_info:
+    cloudwatch_metric_alarm_info:
+        alarm_names:
+            - my-test-alarm-1
+            - my-test-alarm-2
+
+'''
+
+RETURN = '''
```

Comment:
Not sure if we want to have the returned data in the same way the cloudwatch_metric_alarm modules does https://github.com/ansible-collections/community.aws/blob/main/plugins/modules/cloudwatch_metric_alarm.py

@tremble @jillr

Reply:
that is interesting indeed.
I have followed approach that's present in other `info` modules
examples: [aws_az_info](https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/aws_az_info.py#L181), [ec2_ami_info](https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/ec2_ami_info.py#L262), [ec2_instance_info](https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/ec2_instance_info.py#L559).
but certainly open to the idea you are suggesting @alinabuzachis.

Reply:
It looks like the differences are that `cloudwatch_metric_alarm` changes AlarmName to `name`, does not return several keys, and doesn't return a list. Since the info module could return data about multiple alarms I think it makes sense to keep this a list of dicts, and it's better to return all the keys the API gives us instead of a hard-coded subset.  

For the `name` vs `alarm_name` key, it would be ideal if they were the same but I'd prefer to keep the same key that AWS does here ('alarm_name`).

Reply:
https://github.com/ansible-collections/amazon.aws/blob/main/docs/docsite/rst/dev_guidelines.rst#returning-values

If botocore returns alarm_name, we should return alarm_name.